sdk/core/System.ClientModel/tests/client/ModelReaderWriter/Models/DiscriminatorSet/BaseModel.cs

```diff
@@ -9,35 +9,38 @@
 namespace System.ClientModel.Tests.Client.ModelReaderWriterTests.Models
 {
     [PersistableModelProxy(typeof(UnknownBaseModel))]
-    public abstract class BaseModel : IJsonModel<BaseModel>
+    public abstract class BaseModel : IJsonModel<BaseModel?>
     {
-        private Dictionary<string, BinaryData> _rawData;
+        private readonly Dictionary<string, BinaryData> _rawData;
 
-        public static implicit operator BinaryContent(BaseModel baseModel)
+        protected internal BaseModel(Dictionary<string, BinaryData>? rawData)
+        {
+            _rawData = rawData ?? new Dictionary<string, BinaryData>();
+        }
+
+        internal Dictionary<string, BinaryData> GetRawData() => _rawData;
+
+        public static implicit operator BinaryContent?(BaseModel? baseModel)
         {
             if (baseModel == null)
             {
                 return null;
             }
 
-            return BinaryContent.Create(baseModel, ModelReaderWriterHelper.WireOptions);
+            return BinaryContent.Create(new NonNullable(baseModel), ModelReaderWriterHelper.WireOptions);
```

Comment:
This is a possible pattern for calling BinaryContent.Create and ModelReaderWriter.Create given the non-nullable constraint on these types.